scrapy_autounit/utils.py

```diff
@@ -60,29 +60,28 @@ def create_dir(path, parents=False, exist_ok=False):
             raise
 
 
-def get_or_create_fixtures_dir(base_path, spider_name, callback_name):
-    create_tests_tree(base_path, spider_name, callback_name)
-    fixtures_dir = base_path / 'fixtures' / spider_name / callback_name
-    create_dir(fixtures_dir, parents=True, exist_ok=True)
-    return fixtures_dir
-
-
-def create_tests_tree(base_path, spider_name, callback_name):
-    tests_dir = base_path / 'tests' / spider_name / callback_name
-    create_dir(tests_dir, parents=True, exist_ok=True)
-    (base_path / '__init__.py').touch()
-    (base_path / 'tests' / '__init__.py').touch()
-    (base_path / 'tests' / spider_name / '__init__.py').touch()
-    (base_path / 'tests' / spider_name / callback_name / '__init__.py').touch()
-
-
-def add_sample(index, fixtures_dir, data):
-    filename = 'fixture%s.bin' % str(index)
-    path = fixtures_dir / filename
+def get_or_create_test_dir(base_path, spider_name, callback_name, extra=None):
+    components = [base_path, 'tests', spider_name]
+    if extra:
+        components.append(extra)
+    components.append(callback_name)
+    test_dir = None
+    for component in components:
+        test_dir = test_dir / component if test_dir else component
+        test_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
This fails in Python versions prior to 3.5 since `exist_ok` parameter was introduced in that version.
Maybe we could replace it with `create_dir(test_dir, parents=True, exist_ok=True)`

Reply:
Got it!  Makes sense.